Date: Thu, 21 Jun 2018 19:29:13 +0200
Subject: [PATCH 2/2] Bump Dagger to 2.14.1 to support Java 9

Closes #275
---
 closure/repositories.bzl | 41 +++++++++++++++++++++++++++++------------
 1 file changed, 29 insertions(+), 12 deletions(-)

diff --git a/closure/repositories.bzl b/closure/repositories.bzl
index 2816fb6..fe63b6f 100644
--- closure/repositories.bzl
+++ closure/repositories.bzl
@@ -35,6 +35,7 @@ def closure_repositories(
     omit_com_google_dagger=False,
     omit_com_google_dagger_compiler=False,
     omit_com_google_dagger_producers=False,
+    omit_com_google_dagger_spi=False,
     omit_com_google_errorprone_error_prone_annotations=False,
     omit_com_google_errorprone_javac_shaded=False,
     omit_com_google_guava=False,
@@ -101,6 +102,8 @@ def closure_repositories(
     com_google_dagger_compiler()
   if not omit_com_google_dagger_producers:
     com_google_dagger_producers()
+  if not omit_com_google_dagger_spi:
+    com_google_dagger_spi()
   if not omit_com_google_errorprone_error_prone_annotations:
     com_google_errorprone_error_prone_annotations()
   if not omit_com_google_errorprone_javac_shaded:
@@ -418,10 +421,10 @@ def com_google_common_html_types_html_proto():
 def com_google_dagger():
   java_import_external(
       name = "com_google_dagger",
-      jar_sha256 = "8b7806518bed270950002158934fbd8281725ee09909442f2f22b58520b667a7",
+      jar_sha256 = "374cfee26c9c93f44caa1946583c9edc135bb9a42838476522551ec46aa55c7c",
       jar_urls = [
-          "https://mirror.bazel.build/repo1.maven.org/maven2/com/google/dagger/dagger/2.9/dagger-2.9.jar",
-          "https://repo1.maven.org/maven2/com/google/dagger/dagger/2.9/dagger-2.9.jar",
+          "https://mirror.bazel.build/repo1.maven.org/maven2/com/google/dagger/dagger/2.14.1/dagger-2.14.1.jar",
+          "https://repo1.maven.org/maven2/com/google/dagger/dagger/2.14.1/dagger-2.14.1.jar",
       ],
       licenses = ["notice"],  # Apache 2.0
       deps = ["@javax_inject"],
@@ -441,17 +444,20 @@ def com_google_dagger():
 def com_google_dagger_compiler():
   java_import_external(
       name = "com_google_dagger_compiler",
-      jar_sha256 = "afe356def27710db5b60cad8e7a6c06510dc3d3b854f30397749cbf0d0e71315",
+      jar_sha256 = "ff16d55273e375349537fc82292b00de04d8a2caca2d4aa6c642692b1a68194d",
       jar_urls = [
-          "https://mirror.bazel.build/repo1.maven.org/maven2/com/google/dagger/dagger-compiler/2.9/dagger-compiler-2.9.jar",
-          "https://repo1.maven.org/maven2/com/google/dagger/dagger-compiler/2.9/dagger-compiler-2.9.jar",
+          "https://mirror.bazel.build/repo1.maven.org/maven2/com/google/dagger/dagger-compiler/2.14.1/dagger-compiler-2.14.1.jar",
+          "https://repo1.maven.org/maven2/com/google/dagger/dagger-compiler/2.14.1/dagger-compiler-2.14.1.jar",
       ],
       licenses = ["notice"],  # Apache 2.0
       deps = [
           "@com_google_code_findbugs_jsr305",
           "@com_google_dagger//:runtime",
           "@com_google_dagger_producers//:runtime",
+          "@com_google_dagger_spi",
           "@com_google_guava",
+          "@com_google_java_format",
+          "@com_squareup_javapoet",
       ],
       extra_build_file_content = "\n".join([
           "java_plugin(",
@@ -471,10 +477,10 @@ def com_google_dagger_compiler():
 def com_google_dagger_producers():
   java_import_external(
       name = "com_google_dagger_producers",
-      jar_sha256 = "b452dc1b95dd02f6272e97b15d1bd35d92b5f484a7d69bb73887b6c6699d8843",
+      jar_sha256 = "96f950bc4b94d013b0c538632a4bc630f33eda8b01f63ae752b76c5e48783859",
       jar_urls = [
-          "https://mirror.bazel.build/repo1.maven.org/maven2/com/google/dagger/dagger-producers/2.9/dagger-producers-2.9.jar",
-          "https://repo1.maven.org/maven2/com/google/dagger/dagger-producers/2.9/dagger-producers-2.9.jar",
+          "https://mirror.bazel.build/repo1.maven.org/maven2/com/google/dagger/dagger-producers/2.14.1/dagger-producers-2.14.1.jar",
+          "https://repo1.maven.org/maven2/com/google/dagger/dagger-producers/2.14.1/dagger-producers-2.14.1.jar",
       ],
       licenses = ["notice"],  # Apache 2.0
       deps = [
@@ -495,6 +501,17 @@ def com_google_dagger_producers():
       ]),
   )
 
+def com_google_dagger_spi():
+  java_import_external(
+      name = "com_google_dagger_spi",
+      jar_sha256 = "6a20d6c6620fefe50747e9e910e0d0c178cf39d76b67ccffb505ac9a167302cb",
+      jar_urls = [
+          "https://mirror.bazel.build/repo1.maven.org/maven2/com/google/dagger/dagger-spi/2.14.1/dagger-spi-2.14.1.jar",
+          "https://repo1.maven.org/maven2/com/google/dagger/dagger-spi/2.14.1/dagger-spi-2.14.1.jar",
+      ],
+      licenses = ["notice"],  # Apache 2.0
+  )
+
 def com_google_errorprone_error_prone_annotations():
   java_import_external(
       name = "com_google_errorprone_error_prone_annotations",
@@ -749,10 +766,10 @@ def com_ibm_icu_icu4j():
 def com_squareup_javapoet():
   java_import_external(
       name = "com_squareup_javapoet",
-      jar_sha256 = "2f671d5f056f04922feff78dd60c34979fc9863b16ad706551a9b68842c1a3d0",
+      jar_sha256 = "5bb5abdfe4366c15c0da3332c57d484e238bd48260d6f9d6acf2b08fdde1efea",
       jar_urls = [
-          "https://mirror.bazel.build/repo1.maven.org/maven2/com/squareup/javapoet/1.7.0/javapoet-1.7.0.jar",
-          "https://repo1.maven.org/maven2/com/squareup/javapoet/1.7.0/javapoet-1.7.0.jar",
+          "https://mirror.bazel.build/repo1.maven.org/maven2/com/squareup/javapoet/1.9.0/javapoet-1.9.0.jar",
+          "https://repo1.maven.org/maven2/com/squareup/javapoet/1.9.0/javapoet-1.9.0.jar",
       ],
       licenses = ["notice"],  # Apache 2.0
   )
-- 
2.16.3

