diff --git a/polygerrit-ui/app/.eslintrc.js b/polygerrit-ui/app/.eslintrc.js
index cd1055b..a131f56 100644
--- a/polygerrit-ui/app/.eslintrc.js
+++ b/polygerrit-ui/app/.eslintrc.js
@@ -316,6 +316,13 @@
           'error',
           {ignoreArrowShorthand: true},
         ],
+        // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/strict-boolean-expressions.md
+        '@typescript-eslint/strict-boolean-expressions': ['error',
+          {
+            allowNullableString: true,
+            allowNullableBoolean: true,
+          },
+        ],
         '@typescript-eslint/no-unused-vars': [
           'error',
           {argsIgnorePattern: '^_'},
@@ -347,6 +354,7 @@
       rules: {
         '@typescript-eslint/no-explicit-any': 'off',
         '@typescript-eslint/require-await': 'off',
+        '@typescript-eslint/strict-boolean-expressions': 'off',
       },
     },
     {
