--- a/node_modules/twinkie/src/app/index.js
+++ b/node_modules/twinkie/src/app/index.js
@@ -250,7 +250,7 @@ twinkie --tsconfig tsconfig.json --outdir output_dir [--files file_list] [--outt
                 incremental: false,
                 noEmit: true,
             },
-            files: [...allProgramFilesNames, generatedFiles],
+            files: [...allProgramFilesNames, ...generatedFiles],
         };
         fs.writeFileSync(cmdLineOptions.outputTsConfig, JSON.stringify(tsconfigContent, null, 2));
     }
